plugins/modules/ec2_metadata_facts.py

```diff
@@ -243,8 +243,12 @@
             description: Data that can be used by other parties to verify its origin and authenticity.
             type: str
             sample: ""
+        ansible_ec2_instance_life_cycle:
+            description: The life cycle of the instance.
```

Comment:
```suggestion
            description: The purchasing option of the instance.
```
Based on the AWS docs:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html

It seems a little counter-intuitive, but it's a description of what life-cycle the instance will follow rather the the current state